lib/src/migrators/division.dart

```diff
@@ -56,128 +56,141 @@ class _DivisionMigrationVisitor extends MigrationVisitor {
+  /// as the aggressive option is enabled.
+  bool _returnsNumbers(BinaryOperator operator) =>
       operator == BinaryOperator.times ||
       operator == BinaryOperator.modulo ||
-      isAggressive &&
+      !isPessimistic &&
           (operator == BinaryOperator.plus || operator == BinaryOperator.minus);
 
   /// Returns true if we assume that [operator] always operators on numbers.
   ///
-  /// This is always true for *, %, <, <=, >, and >=, and it's true for +, -,
-  /// ==, and != as long as the aggressive option is enabled.
-  bool operatesOnNumbers(BinaryOperator operator) =>
-      returnsNumbers(operator) ||
+  /// This is always true for `*`, `%`, `<`, `<=`, `>`, and `>=`, and it's true
+  /// for `+`, `-`, `==`, and `!=` as long as the aggressive option is enabled.
+  bool _operatesOnNumbers(BinaryOperator operator) =>
+      _returnsNumbers(operator) ||
       operator == BinaryOperator.lessThan ||
       operator == BinaryOperator.lessThanOrEquals ||
       operator == BinaryOperator.greaterThan ||
       operator == BinaryOperator.greaterThanOrEquals ||
-      isAggressive &&
+      !isPessimistic &&
           (operator == BinaryOperator.equals ||
               operator == BinaryOperator.notEquals);
 
   /// Returns true if [node] should be treated as division and migrated.
   ///
   /// Warns if division is allowed but it's unclear whether or not all types
   /// are numeric.
-  bool shouldMigrate(BinaryOperationExpression node) {
-    if (!_isDivisionAllowed && onlySlash(node)) {
-      return false;
-    }
-    if (hasNonNumber(node)) return false;
-    if (_expectsNumericResult || _allNumeric(node)) return true;
+  bool _shouldMigrate(BinaryOperationExpression node) {
+    if (!_isDivisionAllowed && _onlySlash(node)) return false;
+    if (_isDefinitelyNotNumber(node)) return false;
+    if (_expectsNumericResult || _isDefinitelyNumber(node)) return true;
     warn("Could not determine whether this is division", node.span);
     return false;
   }
 
   /// Returns true if [node] is entirely composed of number literals and slash
   /// operations.
-  bool onlySlash(Expression node) {
+  bool _onlySlash(Expression node) {
     if (node is NumberExpression) return true;
     if (node is BinaryOperationExpression) {
       return node.operator == BinaryOperator.dividedBy &&
-          onlySlash(node.left) &&
-          onlySlash(node.right);
+          _onlySlash(node.left) &&
+          _onlySlash(node.right);
     }
     return false;
   }
 
   /// Returns true if [node] is believed to always evaluate to a number.
-  bool _allNumeric(Expression node) {
+  bool _isDefinitelyNumber(Expression node) {
     if (node is NumberExpression) return true;
-    if (node is ParenthesizedExpression) return _allNumeric(node.expression);
-    if (node is UnaryOperationExpression) return _allNumeric(node.operand);
-    if (node is FunctionExpression) return isAggressive;
+    if (node is ParenthesizedExpression) {
```

Comment:
Style nit: since these are getting expanded out into multiline `if`s now, I might use `else if` even though they automatically short-circuit just to reduce the number of extra lines.